examples/pytorch/evolveGCN/model.py

```diff
@@ -0,0 +1,201 @@
+import math
+import torch
+import torch.nn as nn
+from dgl.nn.pytorch import GraphConv
+from torch.nn.parameter import Parameter
+
+
+class MatGRUCell(torch.nn.Module):
+    """
+    GRU cell for matrix, similar to the official code.
+    Please refer to section 3.4 of the paper for the formula.
+    """
+
+    def __init__(self, in_feats, out_feats):
+        super().__init__()
+        self.update = MatGRUGate(in_feats,
+                                 out_feats,
+                                 torch.nn.Sigmoid())
+
+        self.reset = MatGRUGate(in_feats,
+                                out_feats,
+                                torch.nn.Sigmoid())
+
+        self.htilda = MatGRUGate(in_feats,
+                                 out_feats,
+                                 torch.nn.Tanh())
+
+    def forward(self, prev_Q, z_topk=None):
+        if z_topk is None:
+            z_topk = prev_Q
+
+        update = self.update(z_topk, prev_Q)
+        reset = self.reset(z_topk, prev_Q)
+
+        h_cap = reset * prev_Q
+        h_cap = self.htilda(z_topk, h_cap)
+
+        new_Q = (1 - update) * prev_Q + update * h_cap
+
+        return new_Q
+
+
+class MatGRUGate(torch.nn.Module):
+    """
+    GRU gate for matrix, similar to the official code.
+    Please refer to section 3.4 of the paper for the formula.
+    """
+
+    def __init__(self, rows, cols, activation):
+        super().__init__()
+        self.activation = activation
+        self.W = Parameter(torch.Tensor(rows, rows))
+        self.reset_param(self.W)
+
+        self.U = Parameter(torch.Tensor(rows, rows))
+        self.reset_param(self.U)
+
+        self.bias = Parameter(torch.zeros(rows, cols))
+
+    def reset_param(self, t):
+        # Initialize based on the number of columns
+        stdv = 1. / math.sqrt(t.size(1))
+        t.data.uniform_(-stdv, stdv)
```

Comment:
Is this just `torch.nn.init.xavier_uniform_`?

Reply:
done.